lib/kafkaClient.js

```diff
@@ -695,6 +747,81 @@ KafkaClient.prototype.clearCallbackQueue = function (socket, error) {
   delete this.cbqueue[socketId];
 };
 
+/**
+ * Fetches metadata for brokers and cluster.
+ * This includes an array containing each node (id, host and port).
+ * Depending on kafka version, additional cluster information is available (controller id).
+ * @param {loadMetadataCallback} cb Function to call once metadata is loaded.
+ */
+KafkaClient.prototype.loadMetadata = function (callback) {
+  this.loadMetadataForTopics(null, callback);
+};
+
+/**
+ * Fetches metadata for brokers and cluster.
+ * This includes an array containing each node (id, host and port). As well as an object
+ * containing the topic name, partition, leader number, replica count, and in sync replicas per partition.
+ * Depending on kafka version, additional cluster information is available (controller id).
+ * @param {Array} topics List of topics to fetch metadata for. An empty array ([]) will fetch all topics.
+ * @param {loadMetadataCallback} callback Function to call once metadata is loaded.
+ */
+KafkaClient.prototype.loadMetadataForTopics = function (topics, callback) {
+  var broker = this.brokerForLeader();
+
+  if (!broker || !broker.socket || broker.socket.error || broker.socket.destroyed) {
+    return callback(new errors.BrokerNotAvailableError('Broker not available'));
+  }
+
+  const ensureBrokerReady = (broker, cb) => {
+    if (!broker.isReady()) {
+      logger.debug('missing apiSupport waiting until broker is ready...');
+      this.waitUntilReady(broker, cb);
+    } else {
+      cb(null);
+    }
+  };
+
+  async.series([
+    cb => {
+      ensureBrokerReady(broker, cb);
+    },
+    cb => {
+      var correlationId = this.nextId();
+      var supportedCoders = getSupportedForRequestType(broker, 'metadata');
+      var request = supportedCoders.encoder(this.clientId, correlationId, topics);
+
+      this.queueCallback(broker.socket, correlationId, [supportedCoders.decoder, cb]);
+      broker.write(request);
+    }
+  ], (err, result) => {
+    callback(err, result[1]);
+  });
+};
+
+/**
+ * Creates one or more topics.
+ * @param {Array} topics Array of topics with partition and replication factor to create.
+ * @param {createTopicsCallback} callback Function to call once operation is completed.
+ */
+KafkaClient.prototype.createTopics = function (topics, callback) {
+  // Calls with [string, string, ...] are forwarded to support previous versions
+  if (topics.every(t => typeof t === 'string')) {
+    return Client.prototype.createTopics.apply(this, arguments);
+  }
+
+  this.getController((error, broker) => {
+    if (error) {
+      return callback(error);
+    }
+
+    const correlationId = this.nextId();
+    const request = protocol.encodeCreateTopicRequest(this.clientId, correlationId, topics, this.options.requestTimeout);
+
+    this.queueCallback(broker.socket, correlationId, [protocol.decodeCreateTopicResponse, callback]);
```

Comment:
Do we want to clear the cached controller createTopics again when we get a `NOT_CONTROLLER` error?

Reply:
Good idea. In `Client` we have `sendGroupRequest` so possibly add a `sendControllerRequest`?